misc/ci_setup.py

```diff
@@ -188,7 +188,7 @@ def run(self):
                 "sourceinspect",
                 "pytest",
                 "pytest-xdist",
-                "pytest-rerunfailures",
+                "pytest-rerunfailures<16.0",
```

Comment:
Can we please add a comment with the motivation for pinning, possibly with a link to the relevant GitHub issue, for future reference?